pmd-scala/src/main/java/net/sourceforge/pmd/cpd/ScalaTokenizer.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.cpd;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Properties;
+
+import org.apache.commons.lang3.StringUtils;
+
+import net.sourceforge.pmd.lang.LanguageRegistry;
+import net.sourceforge.pmd.lang.LanguageVersion;
+import net.sourceforge.pmd.lang.scala.ScalaLanguageHandler;
+import net.sourceforge.pmd.lang.scala.ScalaLanguageModule;
+
+import scala.collection.JavaConverters;
+import scala.meta.Dialect;
+import scala.meta.inputs.Input;
+import scala.meta.internal.tokenizers.ScalametaTokenizer;
+
+/**
+ * Scala Tokenizer class. Uses the Scala Meta Tokenizer.
+ */
+public class ScalaTokenizer implements Tokenizer {
+
+    /**
+     * Denotes the version of the scala dialect to use. Based on the values in
+     * {@linkplain ScalaLanguageModule#getVersions()}
+     */
+    public static final String SCALA_VERSION_PROPERTY = "scala_version";
```

Comment:
I am not a language expert in scala in particular, but 2.13 seems to have added interpolated strings. See the section just above Escape Sequences in 2.12 and 2.13: 
https://www.scala-lang.org/files/archive/spec/2.12/01-lexical-syntax.html#escape-sequences
https://www.scala-lang.org/files/archive/spec/2.13/01-lexical-syntax.html#escape-sequences

To your point, I have this default to latest, but it can be set backward to the more accurate version if needed. I'm happy to just set it to latest and not have it configurable as it seems that the spec is only additive.

Reply:
Too bad, that the changelog doesn't seem to be updated anymore.... https://www.scala-lang.org/files/archive/spec/2.13/15-changelog.html